### PR TITLE
Adds support for filtering by name in /list

### PR DESCRIPTION
**Pull request by dposada**

### CODE REVIEW

File: integration/tests/cook/test_basic.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def test_failing_submit(self):
 
     def test_progress_update_submit(self):
         command = 'echo "progress: 25 Twenty-five percent"; sleep 1; exit 0'
-        job_uuid, resp = util.submit_job(self.cook_url, command=command, max_runtime=5000)
+        job_uuid, resp = util.submit_job(self.cook_url, command=command, max_runtime=60000)
```
Comment (pschorf) [inline integration/tests/cook/test_basic.py]: Why did this need to change? 12x seems like a big bump

File: scheduler/src/cook/mesos/api.clj [hunk 2]
```diff
@@ -1585,25 +1609,35 @@
...
-                                    ::user user
-                                    ::since-hours-ago (parse-int-default since-hours-ago 24)
-                                    ::start-ms (parse-long-default start-ms nil)
-                                    ::limit (parse-int-default limit Integer/MAX_VALUE)
-                                    ::end-ms (parse-long-default end-ms (System/currentTimeMillis))}]
-                            (catch NumberFormatException e
-                              [true {::error (.toString e)}]))
-                          [true {::error (str "unsupported state in " state ", must be one of: " allowed-list-states)}]))
-                      [true {::error "must supply the state and the user name"}])))
+                                             (get-in ctx [:request :body-params])))
```
Comment (pschorf) [inline scheduler/src/cook/mesos/api.clj]: 👏

File: scheduler/src/cook/mesos/api.clj [hunk 3]
```diff
@@ -1576,6 +1580,26 @@
...
+  [name]
+  (re-pattern (-> name
+                  (str/replace #"\." "\\\\.")
+                  (str/replace #"\*+" ".*"))))
+
+(defn name-filter-str->name-filter-fn
+  "Returns a name-filtering function (or nil) given a user-provided name filter string"
+  [name]
+  (if name
+    #(re-matches (name-filter-str->name-filter-pattern name) %)
```
Comment (pschorf) [inline scheduler/src/cook/mesos/api.clj]: Not sure if this actually matters but would it make a difference to compile the regex once and re-use it?

File: integration/tests/cook/test_basic.py (hunk 1 above)
Comment (dposada) [reply integration/tests/cook/test_basic.py]: On Travis, the test was failing due to "task max runtime exceeded"

File: scheduler/src/cook/mesos/api.clj (hunk 3 above)
Comment (dposada) [reply scheduler/src/cook/mesos/api.clj]: Good idea, done

**Comment by dposada** [comment]
Build is green now

**Pull request merged**